### ISSUE TITLE
test-tls-server often fails and causes hangs

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#13238]
I see this failure a lot in semaphore, e. g. [here](https://semaphoreci.com/cockpit-project/cockpit/branches/pull-request-13236/builds/4):
```
test-tls-server: connect(/tmp/server.wssock.IY98B0/http.sock) failed: No such file or directory
[...]
ERROR:src/tls/test-server.c:157:recv_reply: assertion failed (len >= 100): (0 >= 100)
SKIP: test-tls-server Bail out! ERROR:src/tls/test-server.c:157:recv_reply: assertion failed (len >= 100): (0 >= 100)
Aborted
Could not connect: No such file or directory
execution expired
```

another failure:
```
PASS: test-tls-server 10 /server/tls/client-cert-expired
**
ERROR:src/tls/test-server.c:699:test_tls_client_cert_parallel: assertion failed (status == 0): (1141136 == 0)
SKIP: test-tls-server Bail out! ERROR:src/tls/test-server.c:699:test_tls_client_cert_parallel: assertion failed (status == 0): (1141136 == 0)
Aborted
**
ERROR:src/tls/test-server.c:626:test_tls_client_cert_parallel: assertion failed (gnutls_handshake (sessions[i]) == GNUTLS_E_SUCCESS): (-54 == 0)
SKIP: test-tls-server Bail out! ERROR:src/tls/test-server.c:626:test_tls_client_cert_parallel: assertion failed (gnutls_handshake (sessions[i]) == GNUTLS_E_SUCCESS): (-54 == 0)
execution expired
```

Not sure how long semaphore retains the log, so I uploaded the [full](https://piware.de/tmp/gcc.log) [log](https://piware.de/tmp/i386.log) to my server for now.

**Comment by martinpitt** [comment]
I saw [another race](https://semaphoreci.com/cockpit-project/cockpit/branches/pull-request-13248/builds/8):
```
PASS: test-tls-server 4 /server/no-tls/many-serial
**
ERROR:src/tls/test-certfile.c:199:test_certfile_multithreaded: assertion failed (rmdir (dirname) == 0): (-1 == 0)
SKIP: test-tls-certfile Bail out! ERROR:src/tls/test-certfile.c:199:test_certfile_multithreaded: assertion failed (rmdir (dirname) == 0): (-1 == 0)
Aborted (core dumped)
ERROR: test-tls-certfile process failed: 134
```

**Comment by allisonkarlitskaya** [comment]
> I saw [another race](https://semaphoreci.com/cockpit-project/cockpit/branches/pull-request-13248/builds/8):
> 
> ```
> PASS: test-tls-server 4 /server/no-tls/many-serial
> **
> ERROR:src/tls/test-certfile.c:199:test_certfile_multithreaded: assertion failed (rmdir (dirname) == 0): (-1 == 0)
> SKIP: test-tls-certfile Bail out! ERROR:src/tls/test-certfile.c:199:test_certfile_multithreaded: assertion failed (rmdir (dirname) == 0): (-1 == 0)
> Aborted (core dumped)
> ERROR: test-tls-certfile process failed: 134
> ```

This one is #13260

**Comment by allisonkarlitskaya** [comment]
Another:

```
ERROR:src/tls/test-server.c:680:test_tls_client_cert_parallel: assertion failed (access (tc->cert_file_path, F_OK) == 0): (-1 == 0)
**
ERROR:src/tls/test-server.c:697:test_tls_client_cert_parallel: assertion failed (status == 0): (6 == 0)
Aborted

```

**Comment by allisonkarlitskaya** [comment]
The rest of this was probably addressed with #13268.
